src/lib/utils/index.ts

```diff
@@ -28,17 +15,48 @@ export function clone(item) {
   return JSON.parse(JSON.stringify(item));
 }
 
-export function lsSet(key: string, value: any) {
-  return localStorage.setItem(`${pkg.name}.${key}`, JSON.stringify(value));
+function lsAddVersion(value: any, version: string) {
+  return {
+    data: value,
+    _version: version
+  };
 }
 
-export function lsGet<T = any>(key: string, fallback?: any): T {
-  const item = localStorage.getItem(`${pkg.name}.${key}`);
-  return jsonParse(item, fallback);
+function lsGetKey(key: string) {
+  return `${pkg.name}.${key}`;
+}
+
+export function lsSet(key: string, value: any, version?: string) {
+  const data = version != null ? lsAddVersion(value, version) : value;
+
+  return localStorage.setItem(lsGetKey(key), JSON.stringify(data));
+}
+
+export function lsGet<T = any>(
+  key: string,
+  defaultValue: any = null,
+  version?: string
+): T {
+  const rawValue = localStorage.getItem(lsGetKey(key));
+
+  if (rawValue != null) {
+    try {
+      const value = JSON.parse(rawValue);
+
+      if (version != null) {
+        return value._version === version ? value.data : defaultValue;
+      }
+      return value;
+    } catch (e) {
+      return defaultValue;
+    }
+  }
+
+  return defaultValue;
```

Comment:
If provided with `version` it will try to get the data of that version and will always return the `defaultValue` in case of a mismatch (for version upgrades)